### ISSUE TITLE
[Feature Request] Better automatic persistent cache invalidation

### ISSUE THREAD

**Issue by marty1885** [tenstorrent/tt-metal#12752]
**Is your feature request related to a problem? Please describe.**
I pull and build new versions of tt-metal often in order to keep up with the latest changes. However often (about once a week or more) the newly build TTNN will hang and I had to debug. Is the vast majority of cases, the issue stems from my GGML backend enables the persistent cache but (I assume) the cache contains old code and not rebuilt correctly. The easy solution is to delete the cache directory manually and re-run.

However I expect TTNN cache work like OpenCL and Vulkan shader caches where invalidation is automatic. Making debug much easier.

I've attached a screenshot showing deleting the cache fixes the hang for me (without reset from tt-smi_

![image](https://github.com/user-attachments/assets/4555aecd-f97a-4643-8b08-08fff53d39f8)

**Describe the solution you'd like**
Automatically invalidate persistent cache instead of keep using the old one causing hang.

**Describe alternatives you've considered**

**Additional context**

**Comment by ayerofieiev-tt** [comment]
Makes sense! Thank you, Martin! Let us take a look and follow up with next steps/fix.

**Comment by ayerofieiev-tt** [comment]
Persistent cache feature was removed and replaced by better dependency tracking
